lib/pure/bitops.nim

```diff
@@ -727,89 +730,296 @@ proc countTrailingZeroBits*(x: SomeInteger): int {.inline, noSideEffect.} =
 
-proc rotateRightBits*(value: uint16;
-            amount: range[0..16]): uint16 {.inline, noSideEffect.} =
+func rotateRightBits*(value: uint16, shift: range[0..16]): uint16 {.inline.} =
   ## Right-rotate bits in a 16-bits value.
-  ##
-  ## See also:
-  ## * `rotateRightBits proc <#rotateRightBits,uint8,range[]>`_
-  let amount = amount and 15
-  result = (value shr amount) or (value shl ( (-amount) and 15))
+  runnableExamples:
+    doAssert rotateRightBits(0b00111100_11000011'u16, 1) ==
+      0b10011110_01100001'u16
+    doAssert rotateRightBits(0b00111100_11000011'u16, 3) ==
+      0b01100111_10011000'u16
+    doAssert rotateRightBits(0b00111100_11000011'u16, 6) ==
+      0b00001100_11110011'u16
+    doAssert rotateRightBits(0b00111100_11000011'u16, 8) ==
+      0b11000011_00111100'u16
 
-proc rotateRightBits*(value: uint32;
-            amount: range[0..32]): uint32 {.inline, noSideEffect.} =
+  when nimvm:
+    rotr(value, shift.int32)
+  else:
+    when useBuiltinsRotate:
+      when defined(gcc) or defined(icl):
+        builtin_rotr16(value.cushort, shift.cint).uint16
+      elif defined(clang):
+        builtin_rotr16(value.cushort, shift.cushort).uint16
+      elif defined(vcc):
+        builtin_rotr16(value.cushort, shift.cuchar).uint16
+    else:
+      rotr(value, shift.int32)
+
+func rotateRightBits*(value: uint32, shift: range[0..32]): uint32 {.inline.} =
   ## Right-rotate bits in a 32-bits value.
-  ##
-  ## See also:
-  ## * `rotateRightBits proc <#rotateRightBits,uint8,range[]>`_
-  let amount = amount and 31
-  result = (value shr amount) or (value shl ( (-amount) and 31))
+  runnableExamples:
+    doAssert rotateRightBits(0b0000111111110000_1111000000001111'u32, 1) ==
+      0b1000011111111000_0111100000000111'u32
+    doAssert rotateRightBits(0b0000111111110000_1111000000001111'u32, 5) ==
+      0b0111100001111111_1000011110000000'u32
+    doAssert rotateRightBits(0b0000111111110000_1111000000001111'u32, 12) ==
+      0b0000000011110000_1111111100001111'u32
+    doAssert rotateRightBits(0b0000111111110000_1111000000001111'u32, 16) ==
+      0b1111000000001111_0000111111110000'u32
 
-proc rotateRightBits*(value: uint64;
-            amount: range[0..64]): uint64 {.inline, noSideEffect.} =
+  when nimvm:
+    rotr(value, shift.int32)
+  else:
+    when useBuiltinsRotate:
+      when defined(clang):
+        builtin_rotr32(value.cuint, shift.cuint).uint32
+      else:
+        builtin_rotr32(value.cuint, shift.cint).uint32
+    else:
+      rotr(value, shift.int32)
+
+func rotateRightBits*(value: uint64, shift: range[0..64]): uint64 {.inline.} =
   ## Right-rotate bits in a 64-bits value.
-  ##
-  ## See also:
-  ## * `rotateRightBits proc <#rotateRightBits,uint8,range[]>`_
-  let amount = amount and 63
-  result = (value shr amount) or (value shl ( (-amount) and 63))
+  runnableExamples:
+    doAssert rotateRightBits(0b00000000111111111111111100000000_11111111000000000000000011111111'u64, 1) ==
```

Comment:
why not just 1 example (and then move the rest to tbitops)?

there are no surprises/edge cases here AFAIK

ditto in other similar instances

Reply:
I was just maintaining the old pattern. I will leave just one example. However, I don't think it's necessary to move the others to `tbitops`, as the tests there are more than necessary.